local-cli/generator/index.js

```diff
@@ -8,11 +8,41 @@
  */
 'use strict';
 
+var execSync = require('child_process').execSync;
 var fs = require('fs');
 var path = require('path');
+var semver = require('semver')
 var yeoman = require('yeoman-generator');
 var utils = require('../generator-utils');
 
+// Use Yarn if available, it's much faster than the npm client.
+// Return the version of yarn installed on the system, null if yarn is not available.
+function yarnVersionIfAvailable() {
+  let yarnVersion;
+  try {
+    if (process.platform.startsWith('win')) {
+      // TODO implement on Windows
+      return null;
+    } else {
+      // execSync returns a Buffer -> convert to string
+      yarnVersion = (execSync('yarn --version 2>/dev/null').toString() || '').trim();
+    }
+  } catch (error) {
+    return null;
+  }
+  // yarn < 0.16 has a 'missing manifest' bug
+  try {
+    if (semver.gte(yarnVersion, '0.16.0')) {
+      return yarnVersion;
```

Comment:
Also considering all the trouble...
What if we just ship yarn 0.16 with the CLI?